### PR TITLE
59 TextInput extends native input props

### PR DESCRIPTION
**Pull request by wmangimelli**

### CODE REVIEW

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 1]
```diff
@@ -194,6 +195,7 @@ const TextInput = ({
         id={id}
         type={type}
         multiLineIsResizable={multiLineIsResizable}
+        {...rest}
```
Comment (joien1) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:198]: @aVileBroker @wmangimelli What do you think about removing all of the props like `type` as they are implicitly included? 

I'd also recommend renaming `rest` to something more descriptive. Right now, rest refers to all of the React HTML input element props.

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 2]
```diff
@@ -176,24 +154,15 @@ const TextInput = ({
       <InputComponent
         cols={cols}
         rows={rows}
-        aria-label={ariaLabel}
-        placeholder={placeholder}
         onChange={(e: SyntheticEvent) => {
           e.persist();
           onChange(e);
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:159]: looks like onChange is removed above, this should pull from the rest object
```suggestion
          nativeInputHTMLAttributes.onChange(e);
```

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx [hunk 3]
```diff
@@ -176,24 +154,15 @@ const TextInput = ({
...
           debouncedChange(e);
         }}
-        onKeyPress={onKeyPress}
-        onKeyDown={onKeyDown}
-        onKeyUp={onKeyUp}
-        onFocus={onFocus}
-        onBlur={onBlur}
-        onReset={onReset}
-        onInput={onInput}
         value={value || defaultValue}
```
Comment (aVileBroker) [inline packages/hs-react-ui/src/components/TextInput/TextInput.tsx:162]: We shouldn't do this anymore - they'll pass either value or default value and we should respect that and not put both into value

Comment (wmangimelli) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx]: Gotcha, I am going to remove them altogether so they are just passed through.

File: packages/hs-react-ui/src/components/TextInput/TextInput.tsx (hunk 2 above)
Comment (wmangimelli) [reply packages/hs-react-ui/src/components/TextInput/TextInput.tsx:152]: I ended up adding the onChange back to the api, as it wasn't being recognized.

**Pull request closed without merging**